### PR TITLE
Switch Nautilus to use Intel + OpenMPI

### LINKED ISSUE #45: site-specific linalg provider config
As described in https://github.com/NOAA-EMC/spack-stack/issues/40, and partially implemented in #43, we need providers for the linalg packages. Depending on the system, this can be a combination of
```
   blas:: [openblas]
   lapack:: [openblas]
   fftw-api:: [fftw]
```
(this works for mac and other systems)
or everything can be provided by intel-mkl. As described in https://github.com/NOAA-EMC/spack-stack/issues/40, using intel-mkl on macOS leads to errors with Python 3.9; also, installing intel-mkl for clang/gcc+gfortran users is unnecessary burden.

However, for systems using Intel compilers, in particular when performance is important, we should use MKL.

### LINKED ISSUE #633: [INSTALL] intel-openmpi on nautilus
Please add intel-openmpi stack on nautilus.  Using intelmpi-intel on AMD Epyc Rome infiniband machines produces a deadlock with all2all type communications (JCSDA-internal/oops#2191).  The problem has also been reported with ESMF.  Using openmpi instead of intelmpi fixes the problem with ESMF.

**Which software in the stack would you like installed?**
Please provide the name of the software you would like installed on the HPCs.  Ensure the software exists in the list of supported software.  If new software needs to be added to the stack, use the [add_package_request.md](./add_package_request.md) template.

**What is the version/tag of the software?**
Please provide the version/tag of the software to be installed.

**What compilation options would you like set?**
Give a detailed list of compilation options, compiler and MPI flavors and versions, etc.

**Which machines would you like to have the software installed?**
See currently managed HPC machines.

**Any other relevant information that we should know to correctly install the software??**
Please describe in as much detail as possible.

**Additional context**
Add any other context or screenshots about the install request here.

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

This PR switches the Navy Nautilus computer to use OpenMPI instead of Intel-MPI with the Intel compiler. This is to address hanging MPI all-to-all communication as reported briefly in #633.

The PR also switches the providers for `blas` and `lapack` to `intel-oneapi-mkl` instead of `openblas`. This is done in the site config for now, leaving the default as-is (i.e. `openblas`, which means spack-stack does not require any special packages from Intel etc to be downloaded and installed by default). This addresses the issue https://github.com/JCSDA/spack-stack/issues/45.

### Testing

- Compiled and loaded the modules
- Built jedi-bundle and ran ctests, all tests that were expected to pass did pass.
- Will rely on @areinecke and @sking112 for testing with JEDI-Neptune.

### Applications affected

Any environment using spack-stack on Nautilus

### Systems affected

Navy's Nautilus

### Dependencies

Waiting on https://github.com/JCSDA/spack/pull/288

### Issue(s) addressed

Resolves #633
Resolves https://github.com/JCSDA/spack-stack/issues/45

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by areinecke** [comment]
@climbfuji, is this ready to test?  What is the instruction to load the correct module?

**Comment by climbfuji** [comment]
It's line 274 and following in `doc/source/PreConfiguredSites.rst` in this PR.

**Comment by areinecke** [comment]
Can mkl and nco be added to the modules loaded in the jedi-neptune-env/unified-dev?

**Comment by climbfuji** [comment]
> Can mkl and nco be added to the modules loaded in the jedi-neptune-env/unified-dev?

`nco` is straightforward. For `mkl`, it depends on whether you would also like to use `mkl` in all the spack-stack compiled modules. Right now, the default on all platforms is:
```
      providers:
        blas: [openblas]
        fftw-api: [fftw]
        gl: [opengl]
        glu: [openglu]
        jpeg: [libjpeg-turbo]
        lapack: [openblas]
        yacc: [bison]
```
I think we should switch blas and lapack to `mkl` on nautilus to be consistent, but maybe you just want it for compiling NEPTUNE and you don't care about what the spack-stack libraries are using for blas and lapack?

**Comment by areinecke** [comment]
> I think we should switch blas and lapack to `mkl` on nautilus to be consistent, but maybe you just want it for compiling NEPTUNE and you don't care about what the spack-stack libraries are using for blas and lapack?

I agree, we should probably be consistent with mkl across the whole stack.

**Comment by areinecke** [comment]
I am running into some problems with the p4est/sc compiled library using the new openmpi stack.  It's failing on a very basic call to MPI_Comm_Rank.

```
This is NEPTUNE 1.3.2-1335-g68e4ea59a-dirty
Abort: MPI error
Abort: src/sc.c:1270
Abort: Obtained 11 stack frames
Stack 0: libsc.so.0(sc_abort+0xb3) [0x1555554197e3]
Stack 1: libsc.so.0(sc_abort_verbose+0x78) [0x155555419558]
Stack 2: libsc.so.0(sc_init+0x4eb) [0x15555541938b]
Stack 3: neptune_fcst.exe() [0x6171fb]
Stack 4: neptune_fcst.exe() [0x47e447]
Stack 5: neptune_fcst.exe() [0x4e229f]
Stack 6: neptune_fcst.exe() [0x4e81d9]
Stack 7: neptune_fcst.exe() [0x425175]
Stack 8: neptune_fcst.exe() [0x422da2]
Stack 9: libc.so.6(__libc_start_main+0xe5) [0x155547f4bd85]
Stack 10: neptune_fcst.exe() [0x422cae]
```

sc is a library that's part of p4est.  I'm trying to create a reproducer, but am so far unsuccessful.  Will keep trying.

**Comment by areinecke** [comment]
> I am running into some problems with the p4est/sc compiled library using the new openmpi stack. It's failing on a very basic call to MPI_Comm_Rank.

I have a reproducer in ~reinecke/test/run.sh on nautilus.  This mimics the NEPTUNE call to p4est and passes a mpi communicator.  This works for intelmpi but is failing for openmpi.  Digging into this further with @michalakes.

**Comment by areinecke** [comment]
@climbfuji, can w3nco be added to the jedi-neptune-env/unified-dev too.  There is a dependency in ccpp on this.

**Comment by climbfuji** [comment]
Ok, I am creating a laundry list for this, and will start experimenting with an MKL-based stack. But before I reinstall everything it would be good to find out what's wrong with p4est.

**Comment by areinecke** [comment]
> Ok, I am creating a laundry list for this, and will start experimenting with an MKL-based stack. But before I reinstall everything it would be good to find out what's wrong with p4est.

Sorry, didn't report on that.  We figured that out and JEDI/NEPTUNE compile now.  We are still testing to verify that openmpi fixes the deadlock issue.

File: configs/sites/nautilus/packages.yaml [hunk 1]
```diff
@@ -2,21 +2,44 @@ packages:
...
-    - spec: intel-oneapi-mpi@2021.5.1%intel@2021.5.0
-      modules:
-      - intel/mpi/2021.5.1
-      prefix: /p/app/compilers/intel/oneapi
+  #intel-oneapi-mpi:
+  #  externals:
+  #  - spec: intel-oneapi-mpi@2021.5.1%intel@2021.5.0
+  #    modules:
+  #    - intel/mpi/2021.5.1
+  #    prefix: /p/app/compilers/intel/oneapi
```
Comment (srherbener) [inline configs/sites/nautilus/packages.yaml:25]: Should the commented code be deleted? Or is there a reason to keep it. Thanks

Comment (climbfuji) [reply configs/sites/nautilus/packages.yaml:25]: Yes, this needs to be kept. I expect the sysadmins will fix the problem with oneAPI on the Mellanox fabric soon, and then NRL will want to switch back.

Comment (srherbener) [reply configs/sites/nautilus/packages.yaml:25]: Thanks for the explanation. I think it's fine to leave these in.

**Review by srherbener — APPROVED** [review]

**Review by sking112 — APPROVED** [review]

**Pull request merged**